File: AESCrypt/crypt.py
#!/usr/bin/python3

import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad

#message to be encrypted in string
message = "thisissecretmessage"
#convert message to bytes
msgb = message.encode()


##Encryption
#set parameters
#len(key) and len(iv) must be 16bytes
ivs = "bJQSr2jE8253Plt+IdbeFA=="    #in string
iv = base64.b64decode(ivs)          #in bytes
keys = "secretkey16bytes"           #in string
key = keys.encode()                 #in bytes

#create cipher object
cipher = AES.new(key, AES.MODE_CBC, iv)
#encrypt message using the cipher object
ct_bytes = cipher.encrypt(pad(msgb, AES.block_size))
ct = base64.b64encode(ct_bytes).decode()    #convert ciphertext to readable format: base64 encoding and converting to string
print("cipher text is: " + ct)
print("iv is: " + ivs)
print("key is: " + keys)

##Decryption
#create a new cipher object
cipher2 = AES.new(key, AES.MODE_CBC, iv)
#convert readable ciphertext back to binary format, by base64 decoding 
ct_bytes2 = base64.b64decode(ct)
#decrypt ciphertext
pt = unpad(cipher2.decrypt(ct_bytes2), AES.block_size)

#use comma to concatenate bytes
print("The decrypted message is: ", pt)

File: crypt.java
import java.util.*;
import java.util.Base64;
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

public class crypt {
        
    public static void main(String[] str) throws Exception{
        
        // message to be encrypted, encoded as array of bytes
        String messages = "thisismysecretmessage";
        byte[] message = messages.getBytes();
        
        // set parameters
        //IV 16bytes random string in bytes
        byte ivb[] = new byte[16];
        (new Random()).nextBytes(ivb);
        IvParameterSpec iv = new IvParameterSpec(ivb);
        String ivs = Base64.getEncoder().encodeToString(ivb);   
        //Key 16bytes
        String key = "secretkey16bytes";
        byte[] keyb = key.getBytes();
        SecretKeySpec secretKey = new SecretKeySpec(keyb, "AES");
        
        //Encryption
        //create cipher object
        Cipher encrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING");
        encrypter.init(Cipher.ENCRYPT_MODE, secretKey, iv);
        //encrypt
        byte[] encryptedMsgb = encrypter.doFinal(message);
        //change to readable format: (a) byte to Base64 encode and (b) convert to string
        String encryptedMsg = Base64.getEncoder().encodeToString(encryptedMsgb);
        
        //output
        System.out.println("The encrypted message is: " + encryptedMsg);
        System.out.println("The IV is: " + ivs);
        System.out.println("The key is: " + key);

        //decryption
        
        //create cipher object
        Cipher decrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING");
        decrypter.init(Cipher.DECRYPT_MODE, secretKey, iv);
        //decrypt
        //convert encrypted message to bytes: (a) string to bytes then (b) base64 decode bytes
        byte[] encryptMsgb = Base64.getDecoder().decode(encryptedMsg.getBytes());
        String decryptedMsg = new String(decrypter.doFinal(encryptMsgb));
        
        //output
        System.out.println("The decrypted message is: " + decryptedMsg); 
    }
}
